inc/packages/namespace.php

```diff
@@ -363,23 +363,34 @@ function handle_form() : void {
 
 		// Handle file upload.
 		if ( ! empty( $_FILES ) && isset( $_FILES[ $file_field ] ) && ! empty( $_FILES[ $file_field ]['tmp_name'] ) ) {
-			$file = get_package_path( "uploaded-{$type}", $for_network );
-			// phpcs:ignore HM.Security.ValidatedSanitizedInput.InputNotSanitized
-			$result = move_uploaded_file( wp_unslash( $_FILES[ $file_field ]['tmp_name'] ), $file );
-			if ( ! $result ) {
+			// phpcs:ignore HM.Security.ValidatedSanitizedInput
+			$mime_type = \mime_content_type( $_FILES[ $file_field ]['tmp_name'] );
+			if ( $mime_type !== 'text/plain' ) {
 				$errors[] = new WP_Error(
-					'write_package_error',
+					'file_type_incorrect',
 					// translators: %s replaced by search package file path.
-					sprintf( __( 'Could not write search package file to %s', 'altis' ), $file )
+					sprintf( __( 'Detected unsupported file type %, only textual files are supported.', 'altis' ), $mime_type )
```

Comment:
Not sure textual is a word, but just text here is fine
```suggestion
					sprintf( __( 'Detected unsupported file type %, only text files are supported.', 'altis' ), $mime_type )
```